common/src/main/java/com/wynntils/features/user/overlays/NpcDialogueOverlayFeature.java

```diff
@@ -0,0 +1,155 @@
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@FeatureInfo(category = "Overlays")
+public class NpcDialogueOverlayFeature extends UserFeature {
+    private static final Pattern NEW_QUEST_STARTED = Pattern.compile("^§6§lNew Quest Started: §r§e§l(.*)§r$");
+    private String currentDialogue;
+
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    public void onNpcDialogue(NpcDialogEvent e) {
+        String msg = e.getCodedDialog();
+        if (msg != null && NEW_QUEST_STARTED.matcher(msg).find()) {
+            // TODO: Show nice banner notification instead
+            // but then we'd also need to confirm it with a sneak
+            NotificationManager.queueMessage(msg);
+        }
+        currentDialogue = msg;
+        // Cancel the event so the chat fallback does not get it
+        e.setCanceled(true);
+    }
+
+    @SubscribeEvent
+    public void onWorldStateChange(WorldStateEvent e) {
+        currentDialogue = null;
+    }
+
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    private final Overlay npcDialogueOverlay = new NpcDialogueOverlay();
+
+    public class NpcDialogueOverlay extends Overlay {
+        private TextRenderSetting renderSetting;
+
+        public NpcDialogueOverlay() {
+            super(
+                    new OverlayPosition(
+                            0,
+                            0,
+                            VerticalAlignment.Top,
+                            HorizontalAlignment.Center,
+                            OverlayPosition.AnchorSection.Middle),
+                    new GuiScaledOverlaySize(200, 300),
+                    HorizontalAlignment.Left,
+                    VerticalAlignment.Top);
+            updateTextRenderSettings();
+        }
+
+        private void updateTextRenderSettings() {
+            renderSetting = TextRenderSetting.getWithHorizontalAlignment(
+                    this.getWidth() - 5, CommonColors.WHITE, this.getRenderHorizontalAlignment());
+        }
+
+        @Override
+        protected void onConfigUpdate(ConfigHolder configHolder) {
+            updateTextRenderSettings();
+        }
+
+        private void renderDialogue(PoseStack poseStack, String currentDialogue) {
+            List<TextRenderTask> dialogueRenderTask = List.of(new TextRenderTask(currentDialogue, renderSetting));
+
+            float textHeight = FontRenderer.getInstance().calculateRenderHeight(dialogueRenderTask);
+            // Draw a translucent background
+            // TODO: Replace this with a more stylish background?
+            RenderUtils.drawRectBorders(
+                    poseStack,
+                    CommonColors.DARK_GRAY,
+                    this.getRenderX(),
+                    this.getRenderY(),
+                    this.getRenderX() + this.getWidth(),
+                    this.getRenderY() + textHeight + 10,
+                    1,
+                    1.8f);
```

Comment:
I think we can just not render a border for the overlay, it might look better. If you agree to do this, I will approve after this is changed, if you think we should keep this, LMK.

Reply:
Let me check how that looks. I think it might be a bit bland. 

I'll be occupied elsewhere for a few days so it might be quiet for me for a while.